src/nbl/video/IOpenGL_LogicalDevice.h

```diff
@@ -0,0 +1,692 @@
+                core::smart_refctd_ptr<IGPUImageView>* pretval = reinterpret_cast<core::smart_refctd_ptr<IGPUImageView>*>(req.pretval);
+                pretval[0] = core::make_smart_refctd_ptr<COpenGLImageView>(device, &gl, std::move(p.params));
+            }
+                break;
+            case ERT_SAMPLER_CREATE:
+            {
+                auto& p = std::get<SRequestSamplerCreate>(req.params_variant);
+                core::smart_refctd_ptr<IGPUSampler>* pretval = reinterpret_cast<core::smart_refctd_ptr<IGPUSampler>*>(req.pretval);
+                pretval[0] = core::make_smart_refctd_ptr<COpenGLSampler>(device, &gl, p.params);
+            }
+                break;
+            case ERT_RENDERPASS_INDEPENDENT_PIPELINE_CREATE:
+            {
+                auto& p = std::get<SRequestRenderpassIndependentPipelineCreate>(req.params_variant);
+                core::smart_refctd_ptr<IGPURenderpassIndependentPipeline>* pretval = reinterpret_cast<core::smart_refctd_ptr<IGPURenderpassIndependentPipeline>*>(req.pretval);
+                for (uint32_t i = 0u; i < p.count; ++i)
+                    pretval[i] = createRenderpassIndependentPipeline(gl, p.params[i], p.pipelineCache);
+            }
+                break;
+            case ERT_COMPUTE_PIPELINE_CREATE:
+            {
+                auto& p = std::get<SRequestComputePipelineCreate>(req.params_variant);
+                core::smart_refctd_ptr<IGPUComputePipeline>* pretval = reinterpret_cast<core::smart_refctd_ptr<IGPUComputePipeline>*>(req.pretval);
+                for (uint32_t i = 0u; i < p.count; ++i)
+                    pretval[i] = createComputePipeline(gl, p.params[i], p.pipelineCache);
+            }
+                break;
+
+            case ERT_FLUSH_MAPPED_MEMORY_RANGES:
+            {
+                auto& p = std::get<SRequestFlushMappedMemoryRanges>(req.params_variant);
+                for (auto mrng : p.memoryRanges)
+                    gl.extGlFlushMappedNamedBufferRange(static_cast<COpenGLBuffer*>(mrng.memory)->getOpenGLName(), mrng.offset, mrng.length);
+            }
+                break;
+            case ERT_INVALIDATE_MAPPED_MEMORY_RANGES:
+            {
+                gl.glSync.pglMemoryBarrier(gl.CLIENT_MAPPED_BUFFER_BARRIER_BIT); // i think there's no point in calling it number_of_mem_ranges times?
+            }
+                break;
+            }
+
+            req.ready = true;
+            // moving unlock before the switch (but after cb_begin increment) probably wouldnt hurt
+            lock.unlock(); // unlock so that notified thread wont immidiately block again
+            req.cvar.notify_all(); //notify_one() would do as well, but lets call notify_all() in case something went horribly wrong (theoretically not possible) and multiple threads are waiting for single request
+            lock.lock(); // reacquire (must be locked at the exit of this function -- see system::IThreadHandler docs)
```

Comment:
I think `notify_one` is fine as long as its inside the locked section

Reply:
yep, maybe that's the reason for notifying inside the lock, so the threads that wait dont miss the wakeup....

Reply:
fixed